entrypoint.sh

```diff
@@ -7,227 +7,277 @@ set -o pipefail
-  set -e
+	# Searches for a string specific to AWS CF templates
+	if grep -q 'AWSTemplateFormatVersion' "${file}" >/dev/null; then
+		return 1
+	fi
 
-  echo "-------------------------------" >> "$superlinter_logfile"
+	# Search for AWS, Alexa Skills Kit, or Custom Cloud Formation syntax within the file
+	if grep -q -E '(AWS|Alexa|Custom)::' "${file}" >/dev/null; then
+		return 1
+	fi
 
-  if [ "$superlinter_result" -gt 0 ]; then
-    cat "$superlinter_logfile"
-    linter_failed="true"
-    linter_failures+=("super-linter")
-  else
-    linter_successes+=("super-linter")
-  fi
+	return 0
 }
 
 function get_files_matching_filetype() {
-  local filetype="$1"
-  shift
-  local filenames=("$@")
-  matching_files=()
-
-  for file in "${filenames[@]}"; do
-    filename=$(basename "$file")
-    
-    if [[ "$filename" == *"$filetype" ]]; then
-      matching_files+=("$file")
-    fi
-  done
-  echo "${matching_files[@]}"
+	local filenames=("$@")
+	declare -a matching_files=()
+
+	for filetype in "${linter_filetypes[@]}"; do
+		for file in "${filenames[@]}"; do
+			filename=$(basename "${file}")
+			if [[ $filename == *"$filetype" ]]; then
+				if [ "${linter[name]}" == "cfn-lint" ]; then
+					if detect_cloudformation_file "${file}"; then
+						continue
+					fi
+				fi
+				if [ "${linter[name]}" == "kubeconform" ]; then
+					if detect_kubernetes_file "${file}"; then
+						continue
+					fi
+				fi
+				if [ "${linter[name]}" == "actionlint" ]; then
+					local action_path="./.github/workflows/"
+					if [[ "${file}" != "${action_path}"* ]]; then
+						continue
+					fi
+				fi
+				matching_files+=("${file}")
+			fi
+		done
+	done
+
+	echo "${matching_files[@]}"
 }
 
 function lint_files() {
-  if [[ "${linter[filetype]}" = "all" ]]; then
-    cmd="${linter[name]} ${linter[args]}"
-    eval "$cmd" >> "${linter[logfile]}" 2>&1
-    return
-  fi
-  
-  for file in $(get_files_matching_filetype "${linter[filetype]}" "${included[@]}"); do 
-    if [[ "${linter[executor]+x}" ]]; then
-      cmd="${linter[executor]} ${linter[name]} ${linter[args]} ${file}"
-    else
-      cmd="${linter[name]} ${linter[args]} ${file}"
-    fi
-    eval "$cmd" >> "${linter[logfile]}" 2>&1
-  done
+	local linter_args="${linter[args]}"
+	local files_to_lint=""
+
+	if [[ -v "${linter[env]}" && -n "${!linter[env]}" ]]; then
+		linter_args="${!linter[env]}"
+	fi
+
+	for type in "${linter_filetypes[@]}"; do
+		if [[ $type == "all" ]]; then
+			cmd="${linter[name]} $linter_args"
+			eval "$cmd" >>"${linter[logfile]}" 2>&1
+			return
+		fi
+	done
+
+	files_to_lint="$(get_files_matching_filetype "${included[@]}")"
+
+	if [ "${#files_to_lint}" -eq 0 ]; then
+		return
+	fi
+
+	for file in "${files_to_lint[@]}"; do
+		if [[ "${linter[executor]+x}" ]]; then
+			cmd="${linter[executor]} ${linter[name]} $linter_args ${file}"
+		else
+			cmd="${linter[name]} $linter_args ${file}"
+		fi
+
+		echo "$cmd" >>"${linter[logfile]}"
+		eval "$cmd" >>"${linter[logfile]}" 2>&1
```

Comment:
Instead of this eval and echo, can't se just use `tee` to both send it to stdout and to a file?  You just need to do it something like this (untested):
```suggestion
		eval "$cmd" 2>&1 | tee -a "${linter[logfile]}"
```

The `tee -a` appends to an existing file and may not be needed.  If so, you can just remove the `-a`

Reply:
Done